engine/src/main_run.hpp

```diff
@@ -167,6 +168,11 @@ inline void handle_signal(int sig) {
     case SIGABRT:
       abort();
       break;
+    case SIGQUIT:
```

Comment:
- [ ] Why use `SIGQUIT` to signal a reset?

I am not comfortable using this signal for such a purpose. See https://www.gnu.org/software/libc/manual/html_node/Termination-Signals.html

Better would be `SIGUSR1` or `SIGUSR2`.

The bigger question is:

- [ ] Why can't we inject the "trigger" like any other trigger? Why do we need to resort to low-level signalling?

Reply:
Actually this was only for testing,  of course we could trigger this in web ui , but with web server launched the simulation will be rather slow as we already found. 

Anyway here we should not merge this part into master branch, and it will be rather appreciated if  we could have more user-friendly local client to operate everything, just in my opinion.

Reply:
Here my test code may mislead something，overlapping SIGQUIT was only kind of "convenience" to webui actually.

Binding signal may lead to  tricky issues, for example we need to consider what to do if user try again when we are handing previous signal, and "ctrl +\" lead to different behavior when launch cloe in vscode instead of terminal.

I think have rest server and api may be a better solution, so taht user could  trigger actions via GUI or CLI and get corresponding response.